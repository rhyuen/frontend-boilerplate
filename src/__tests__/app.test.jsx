import React from "react";
import ReactDOM from "react-dom";
import App from "../App.jsx";
import {configure, shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({adapter: new Adapter()});

// it("renders without crashing", () => {
//     const div = document.createElement("div");
//     ReactDOM.render(<App/>, div);
//     ReactDOM.unmountComponentAtNode(div);
// });


describe("<App/>", () => {
    it("should render the App", () => {
        const wrapper = shallow(<App/>, {context: {}, disableLifecycleMethods: true});
        console.log(wrapper.debug());
    });
});