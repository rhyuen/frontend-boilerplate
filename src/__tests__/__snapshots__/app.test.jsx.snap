// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should match the snapshot 1`] = `
<div
  className="root"
>
  <p>
    Hi, Just some vague front end code.
  </p>
  <p>
    The Second Element for the FEBP
  </p>
  <div
    className="block__element--modifier"
  >
    A node for the Enzyme .exists() assertion.
  </div>
  <ul>
    <li>
      ONE ITEM
    </li>
    <li>
      TWO ITEM
    </li>
    <li>
      THREE ITEM
    </li>
  </ul>
  <div
    alt="logoishere"
  >
    There may or may not be a logo here.
  </div>
  <div
    className="button-state"
  >
    Unclicked.
  </div>
  <button
    onClick={[Function]}
  >
    ClickMe!
  </button>
  <div>
    <h2 />
    <input
      onChange={[Function]}
      type="text"
    />
  </div>
  <div>
    <span
      className="blue"
    >
      FirstColour
    </span>
  </div>
  <div>
    <span
      data-testid="lifeCycleMethod"
    >
      componentDidMount
    </span>
  </div>
  <Link
    address="https://engadget.com"
    hide={true}
  />
  <Test />
  <Form />
</div>
`;
